### PR TITLE
fix(component-testing): Fix webpack-dev-server deps validation crash

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
This fixes issue https://cypress-io.atlassian.net/browse/CT-391

Actually, there was a small problem related to next v10 that was fixed here, but the weird error is coming from webpack-dev-server peer dependencies validation which was crashing this installation. 

I cleaned it up a little bit to 1) make sure that we always use real peer dependencies 2) reduce the amount of code and add better types. But I doubt that we need this, nodejs will throw an explicit error if dependencies are missing but we are running this validation on each start for each run on CI and as far I know (not 100% sure) `require` under `try { } catch` will always be deoptimized and won't be cached and overall [require](https://kevin.burke.dev/kevin/node-require-is-dog-slow] is a really slow operation. So maybe we can delete this validation? What do you think @JessicaSachs

### CODE REVIEW

File: npm/webpack-dev-server/src/errors.ts [hunk 1]
```diff
@@ -1,66 +1,33 @@
...
-
-    return {
-      prettyNames: acc.prettyNames.join(', '),
-      packageNames: acc.packageNames.join(' '),
-    }
-  }, { prettyNames: [], packageNames: [] })
-}
+import packageJson from '../package.json'
 
 export class MissingPeerDependency extends Error {
```
Comment (dmtrKovalenko) [inline npm/webpack-dev-server/src/errors.ts:3]: I am not sure to do we need this class later @JessicaSachs ? I see that this is `throw` but why? Do we really need to make a custom instance, BTW thanks to throwing we display this error always twice.

File: npm/webpack-dev-server/src/errors.ts [hunk 2]
```diff
@@ -1,66 +1,33 @@
...
+
+    return `❌ Missing ${packagesToInstall}. Please install them with npm or yarn.
 
-    npm i ${this.libs.packageNames} -D
-    yarn add ${this.libs.packageNames} --dev
+    npm i ${packagesToInstall} -D
+    yarn add ${packagesToInstall} --dev
 
-    Updating webpack config is unnecessary
+    P.S. Updating webpack config is unnecessary
```
Comment (agg23) [inline npm/webpack-dev-server/src/errors.ts:22]: nit: Remove this "P.S.". Too informal

**Review by agg23 — DISMISSED** [review]

File: npm/webpack-dev-server/package.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "name": "@cypress/webpack-dev-server",
   "version": "0.0.0-development",
   "description": "Launches Webpack Dev Server for Component Testing",
-  "main": "index.js",
+  "main": "dist/src/index.js",
```
Comment (lmiller1990) [inline npm/webpack-dev-server/package.json:5]: This change is breaking one of the tests (run with `yarn test`)

Comment (lmiller1990) [reply npm/webpack-dev-server/package.json:5]: `dist` now contains an extra layer - a `src` directory - was this intentional?

**Comment by dmtrKovalenko** [comment]
I decided to delete this test that using __NEXT_DATA__ global because it is diving too deep into next's internal implementation. Originally testing components with router usage should be a bad practice, but using more and more internal APIs is even worse because this test will fail for literally all users between next's releases

**Review by lmiller1990 — DISMISSED** [review]

File: npm/webpack-dev-server/src/startServer.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import Debug from 'debug'
 import webpack from 'webpack'
 import WebpackDevServer from 'webpack-dev-server'
-import { StartDevServer } from '.'
+import { StartDevServer } from '@cypress/webpack-dev-server/src'
```
Comment (elevatebart) [inline npm/webpack-dev-server/src/startServer.ts:4]: Why did you have to do this change? It feels backwards.

File: npm/webpack-dev-server/tsconfig.json [hunk 5]
```diff
@@ -4,6 +4,7 @@
     "target": "es2015" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', or 'ESNEXT'. */,
     "module": "commonjs" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */,
     "skipLibCheck": true,
+    "resolveJsonModule": true,
```
Comment (elevatebart) [inline npm/webpack-dev-server/tsconfig.json:7]: this one should not be necessary anymore.
```suggestion
```

**Review by elevatebart — DISMISSED** [review]
I have 2 non blocking comments.
LGTM

**Review by elevatebart — APPROVED** [review]

**Pull request merged**